cerberus/models/pet.py

```diff
@@ -21,6 +21,11 @@
     from . import Booking
 
 
+class PetManager(models.Manager):
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **issue (complexity):** Consider simplifying the Pet model by reducing external dependencies and custom manager complexity.

While the enhancements introduced in this PR, such as the custom `PetManager` and the use of `humanize.naturaldelta` for calculating the pet's age, aim to improve the functionality and user experience, they also add a notable layer of complexity to the `Pet` model. The custom manager, while powerful, might be an over-engineering for the current requirements, especially if the performance gains are minimal with the existing dataset. Similarly, the dependency on `humanize.naturaldelta` for a more readable age format introduces an external library dependency that needs to be maintained and understood.

Moreover, the addition of model properties like `name_with_owner` and `age` increases the model's responsibilities, potentially making it harder to maintain as more logic is added. It's generally beneficial to keep models lean and focused primarily on representing the database structure, with business logic residing in services or utility functions.

To simplify, consider calculating the pet's age directly within the model without an external library. This approach reduces complexity and external dependencies while still providing a clear, human-readable age format. Here's a suggested implementation:

```python
from datetime import datetime

class Pet(models.Model):
    # Existing fields...

    @property
    def age(self) -> str:
        if self.dob:
            delta = datetime.now().date() - self.dob
            years = delta.days // 365
            months = (delta.days % 365) // 30
            if years > 0:
                return f"{years} years, {months} months"
            else:
                return f"{months} months"
        return "Unknown"
```

This method directly calculates the age in a straightforward manner, avoiding additional complexities and dependencies. It's a more maintainable approach that keeps the model's codebase lean and focused.